### DISCUSSION TITLE
traceId printed in log cann't be found in collector, is this because sampler rate is not 1?

### DISCUSSION THREAD

**Discussion by lmssy** [open-telemetry/opentelemetry-java-instrumentation#3629] [Q&A]
I'm using logback MDC to print traceId and spanId in my log as below
![image](https://user-images.githubusercontent.com/17894328/126254335-2f4eb10b-78fd-4736-aecb-cf5740e33025.png)

But I can't find the traceId above in my elasticsearch（I use collector to send trace and span info to elasticsearch）.

I suspect that is because I set sampler rate to 0.3 not 1, dose the logger will write all traces in log but whether a trace is send to collector will be decided by sampler rate, am I right?

**Comment by iNikem** [comment]
Yes. Even if sampler decides that span should be dropped, that span with a valid `SpanContext` is still present in the runtime and is injected into logging contexts.

**Reply by lmssy** [reply]
Thx for your reply！ 

I have another question, I found the format of traceId in log is like "bded79d6f26a10ac52324681471d973a",
but after opentelemetry-collector send to elasticsearch the traceid's format change to "gb525ojhcuMJuZbwXBC0QA==", totally different.

1. Why the traceId format is different in log and collector?

2. Is there any way to transfer  "gb525ojhcuMJuZbwXBC0QA==" to "bded79d6f26a10ac52324681471d973a" ?

**Reply by trask** [reply]
hey @lmssy! this is probably a question for the elasticsearch exporter/backend
